src/reactor/poll_evented.rs

```diff
@@ -107,7 +114,7 @@ impl<E> PollEvented<E> {
     /// The flag indicating that this stream is readable is unset and the
     /// current task is scheduled to receive a notification when the stream is
     /// then again readable.
-    pub fn need_read(&self) {
+    pub fn need_read(&self) -> io::Result<()> {
```

Comment:
Like above, can you add documentation here for why these functions return an error?